src/components/ListItem.jsx

```diff
@@ -12,23 +12,22 @@ export function ListItem({ listToken, item }) {
 		const currentDate = new Date();
 		const currentTime = currentDate.getTime();
 		let timer = currentTime - calcDate;
+
 		if (item.isChecked && timer >= millisecondsIn24hrs) {
-			updateItem(listToken, {
-				id: item.id,
+			updateItem(listToken, item, {
 				isChecked: false,
 				dateLastPurchased: item.dateLastPurchased,
-				totalPurchases: item.totalPurchases,
 			});
 		}
 	}, []);
 
 	const handleChange = useCallback(async () => {
-		if (item.isChecked === false) {
-			await updateItem(listToken, {
+		if (!item.isChecked) {
+			await updateItem(listToken, item, {
 				id: item.id,
-				isChecked: !item.isChecked,
-				dateLastPurchased: new Date().getTime(),
-				totalPurchases: item.totalPurchases + 1,
+				isChecked: true,
+				dateLastPurchased: item.dateLastPurchased,
+				totalPurchases: item.totalPurchases,
```

Comment:
I'm curious why you are passing three parameters to the `updateItem` functions? because it takes only two arguments (`listId`, and `itemData`). so this is probably causing a bug somewhere